src/main/java/org/gridsuite/filter/server/FilterService.java

```diff
@@ -272,7 +276,25 @@ private <I extends Injection<I>> Stream<Injection<I>> getInjectionList(Stream<In
     }
 
     private List<Identifiable<?>> getGeneratorList(Network network, AbstractFilter filter) {
-        return getInjectionList(network.getGeneratorStream().map(gen -> gen), filter).collect(Collectors.toList());
+        if (filter instanceof CriteriaFilter) {
```

Comment:
It's not possible, for CriteriaFilter, to call getInjectionList and then filter the resulting stream by energy source, to avoid duplicating code ?
and for IdentifierListFilter, just call getInjectionList ?

Reply:
yes good point